src/v/cluster/topic_table.cc

```diff
@@ -1528,7 +1561,8 @@ void topic_table::change_partition_replicas(
   const std::vector<model::broker_shard>& new_assignment,
   topic_metadata_item& metadata,
   partition_assignment& current_assignment,
-  model::offset o) {
+  model::offset o,
+  bool is_forced) {
```

Comment:
for force update to work with controller snapshots we'll need to save somewhere in the topic table that we were force-cancelling (e.g. in the reconfiguration state). Also we'll need to update `last_cmd_revision`

Reply:
done.